ios/Classes/SwiftChargebeeFlutterSdkPlugin.swift

```diff
@@ -214,12 +214,28 @@ extension SKProduct {
     func toMap() -> [String: Any?] {
         let map: [String: Any?] = [
             "productId": productIdentifier,
-            "productPrice": price.description,
+            "productPrice": price,
             "productTitle": localizedTitle,
-            "currencyCode": priceLocale.currencyCode
+            "currencyCode": priceLocale.currencyCode,
+            "subscriptionPeriod": subscriptionDuration()
         ]
         return map
     }
+    func subscriptionDuration() -> String {
+        let period:String = {
+            switch self.subscriptionPeriod?.unit {
+            case .day: return "day"
+            case .week: return "week"
+            case .month: return "month"
+            case .year: return "year"
+            case .none: return ""
+            case .some(_): return ""
+            }
+        }()
+        let numOfUnits = self.subscriptionPeriod?.numberOfUnits ?? 0
+        let pluralInPeriod = numOfUnits > 1 ? "s" : ""
+        return String(format: "%d %@%@", arguments: [numOfUnits, period, pluralInPeriod])
```

Comment:
Why combine them in this arbitrary way? It would be much more flexible if these two separate units of information were split up. Splitting them up would also allow your customers to easily add internationalization.

Reply:
Combined the values(period & numOfUnits) to match with StoreKit Duration under subscriptions. will update the PR by providing the data separately as a type number.